### PR TITLE
Scrolling / pagination of query results

### PR DESCRIPTION
**Pull request by tobias93**

### CODE REVIEW

File: src/main/kotlin/io/georocket/http/StoreEndpoint.kt [hunk 1]
```diff
@@ -159,19 +163,26 @@ class StoreEndpoint(override val coroutineContext: CoroutineContext,
...
-    val search = request.getParam("search")
+    try {
+      val path = Endpoint.getEndpointPath(context)
+      val search = request.getParam("search")
+      val scroll = request.getParam("scroll", "false")
+        .lowercase().trim().toBooleanStrictOrNull()
+        ?: throw HttpException(200, "The parameter 'scroll' must be either 'true' or 'false'.")
+      val size = request.getParam("size", "100")
+        .trim().toIntOrNull()?.takeIf { it > 0 }
+        ?: throw HttpException(200, "The parameter 'size' must be a valid integer that is larger than 0.")
```
Comment (michel-kraemer) [inline src/main/kotlin/io/georocket/http/StoreEndpoint.kt:174]: Status code should be 400

File: src/test/kotlin/io/georocket/index/mongodb/MongoDBIndexTest.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package io.georocket.index.mongodb
+
+import io.georocket.index.Index
+import io.georocket.index.IndexTest
+import io.georocket.storage.mongodb.MongoDBTestConnector
+import io.vertx.core.Vertx
+import org.junit.jupiter.api.AfterAll
+
+class MongoDBIndexTest : IndexTest() {
+
```
Comment (michel-kraemer) [inline src/test/kotlin/io/georocket/index/mongodb/MongoDBIndexTest.kt:10]: File indentation should be two spaces

**Review by michel-kraemer — CHANGES_REQUESTED** [review]
Very good! Thanks! I just have a few minor change requests.

File: src/main/kotlin/io/georocket/http/StoreEndpoint.kt [hunk 3]
```diff
@@ -159,19 +163,26 @@ class StoreEndpoint(override val coroutineContext: CoroutineContext,
...
     val response = context.response()
 
-    val path = Endpoint.getEndpointPath(context)
-    val search = request.getParam("search")
+    try {
+      val path = Endpoint.getEndpointPath(context)
+      val search = request.getParam("search")
+      val scroll = request.getParam("scroll", "false")
+        .lowercase().trim().toBooleanStrictOrNull()
+        ?: throw HttpException(200, "The parameter 'scroll' must be either 'true' or 'false'.")
```
[the remark this answers is not in the document]
Comment (tobias93) [reply src/main/kotlin/io/georocket/http/StoreEndpoint.kt:171]: Oups, good catch, that's a typo...

[the remark this answers is not in the document]
Comment (tobias93) [reply src/main/kotlin/io/georocket/http/StoreEndpoint.kt:171]: done

File: src/main/kotlin/io/georocket/http/StoreEndpoint.kt (hunk 1 above)
Comment (tobias93) [reply src/main/kotlin/io/georocket/http/StoreEndpoint.kt:174]: done

File: src/main/kotlin/io/georocket/http/StoreEndpoint.kt [hunk 4]
```diff
@@ -159,19 +163,26 @@ class StoreEndpoint(override val coroutineContext: CoroutineContext,
     val request = context.request()
     val response = context.response()
 
-    val path = Endpoint.getEndpointPath(context)
-    val search = request.getParam("search")
+    try {
+      val path = Endpoint.getEndpointPath(context)
+      val search = request.getParam("search")
+      val scroll = request.getParam("scroll", "false")
```
[the remark this answers is not in the document]
Comment (tobias93) [reply src/main/kotlin/io/georocket/http/StoreEndpoint.kt:169]: done

File: src/test/kotlin/io/georocket/index/IndexTest.kt [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.extension.ExtendWith
+import kotlin.coroutines.coroutineContext
+
+/**
+ * Abstract test implementation for a [Index]
+ */
+@ExtendWith(VertxExtension::class)
+abstract class IndexTest {
+
```
[the remark this answers is not in the document]
Comment (tobias93) [reply src/test/kotlin/io/georocket/index/IndexTest.kt:22]: Done. Will add an .editorconfig file, that way most text editors will automatically pick up the correct formatting settings for the project

File: src/test/kotlin/io/georocket/index/mongodb/MongoDBIndexTest.kt (hunk 2 above)
Comment (tobias93) [reply src/test/kotlin/io/georocket/index/mongodb/MongoDBIndexTest.kt:10]: done

File: src/test/kotlin/io/georocket/index/postgresql/PostgreSQLIndexTest.kt [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import io.georocket.storage.postgresql.PostgreSQLTest
+import io.vertx.core.Vertx
+import io.vertx.pgclient.PgPool
+import org.testcontainers.containers.PostgreSQLContainerProvider
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+
+@Testcontainers
+class PostgreSQLIndexTest: IndexTest(), PostgreSQLTest {
+
```
[the remark this answers is not in the document]
Comment (tobias93) [reply src/test/kotlin/io/georocket/index/postgresql/PostgreSQLIndexTest.kt:14]: done

**Pull request merged**